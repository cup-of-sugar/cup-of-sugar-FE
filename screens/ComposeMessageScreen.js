import * as WebBrowser from "expo-web-browser";
import * as React from "react";
import {
  ScrollView,
  Image,
  StyleSheet,
  Text,
  TextInput,
  TouchableOpacity,
  View
} from "react-native";
import Colors from "../constants/Colors";
import { gql } from "apollo-boost";
import { useQuery } from "@apollo/react-hooks";
import plane from "../assets/images/plane.png";

export default function ComposeMessageScreen(props) {
  return <ComposeForm action={props.action} navigation={props.navigation} />;
}

class ComposeForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = { subject: "", body: "", error: "", sent: false };
  }

  checkInputs = () => {
    !this.state.subject || !this.state.body
      ? this.setState({ error: "Please add a subject and a message!" })
      : this.sendMessage();
  };

  handleChange = (name, value) => {
    this.setState({ [name]: value });
  };

  sendMessage = () => {
    this.setState({ sent: true });
  };

  // startSearch = () => {
  //   const category = this.state.category;
  //   const itemName = this.state.itemName.toLowerCase() || null;
  //   this.props.navigation.navigate("Search Results", {
  //     category: category,
  //     itemName: itemName,
  //     action: "borrow",
  //     userId: this.props.userId
  //   });
  //   this.setState({ category: "", itemName: "", error: "" });
  // };
  //
  // loanNewItem = () => {
  //   const category = this.state.category;
  //   const itemName = this.state.itemName.toLowerCase();
  //   this.props.navigation.navigate("Loan Details", {
  //     category: category,
  //     name: itemName,
  //     action: "lend",
  //     userId: this.props.userId
  //   });
  //   this.setState({ category: "", itemName: "", error: "" });
  // };

  render() {
    return (
      <View style={styles.formContainer}>
        {!this.state.sent ? (
          <>
            <View style={styles.toContainer}>
              <Text style={styles.header}>To: Joe@tigers.com</Text>
            </View>
            <View style={styles.subjectContainer}>
              <TextInput
                style={styles.subjectInput}
                name="subject"
                value={this.state.subject}
                onChangeText={text => this.handleChange("subject", text)}
                placeholder="Subject..."
              />
            </View>
            <View style={styles.textAreaContainer}>
              <TextInput
                style={styles.textArea}
                name="body"
                value={this.state.body}
                multiline={true}
                numberOfLines={40}
                onChangeText={text => this.handleChange("body", text)}
                placeholder="Message..."
              />
            </View>
            <TouchableOpacity
              style={styles.sendButton}
              onPress={this.checkInputs}
            >
              <Text style={styles.sendButtonText}>Send Message</Text>
            </TouchableOpacity>
            <Text style={styles.errorText}>{this.state.error}</Text>
          </>
        ) : (
          <>
            <Text style={styles.sent}>Sent!</Text>
            <Image source={plane} style={styles.plane} />
            <TouchableOpacity
              style={styles.sendButton}
              onPress={() =>
                this.props.navigation.navigate("Home", {
                  action: this.props.action,
                  userId: this.state.userId
                })
              }
            >
              <Text style={styles.sendButtonText}>Home</Text>
            </TouchableOpacity>
          </>
        )}
      </View>
    );
  }
}

const styles = StyleSheet.create({
  errorText: {
    alignSelf: "center",
    color: "red",
    fontSize: 20,
    fontWeight: "bold",
    margin: 10
  },
  formContainer: {
    backgroundColor: "#fff",
    flex: 1,
    padding: 16,
    paddingTop: 30
  },
  toContainer: {
    backgroundColor: Colors.darkBlue,
    borderTopLeftRadius: 7,
    borderTopRightRadius: 7,
    padding: 5
  },
  subjectContainer: {
    backgroundColor: Colors.darkBlue,
    padding: 5
  },
  subjectInput: {
    backgroundColor: "#fff",
    borderColor: "#CCCCCC",
    borderWidth: 1,
    height: 50,
    fontSize: 22,
    paddingHorizontal: 15
  },
  header: {
    color: "#fff",
    fontSize: 23,
    textAlign: "left",
    fontWeight: "bold",
    padding: 7,
    marginLeft: 10
  },
  textInput: {
    borderColor: "#CCCCCC",
    borderWidth: 1,
    height: 50,
    fontSize: 22,
    margin: 10,
    marginTop: 20,
    paddingLeft: 20,
    paddingRight: 20
  },
  sendButton: {
    marginHorizontal: 40,
    marginVertical: 15,
    paddingTop: 20,
    paddingBottom: 20,
    backgroundColor: Colors.lightBlue,
    borderRadius: 10,
    borderWidth: 1,
    borderColor: "#fff"
  },
  sendButtonText: {
    fontSize: 25,
    fontWeight: "bold",
    color: "#fff",
    textAlign: "center"
  },
  textAreaContainer: {
    borderColor: Colors.darkBlue,
    borderWidth: 5,
    borderBottomLeftRadius: 7,
    borderBottomRightRadius: 7,
    borderTopWidth: 0
  },
  textArea: {
    fontSize: 22,
    height: 400,
    justifyContent: "flex-start",
    padding: 15,
    paddingTop: 20
  },
  sent: {
    alignSelf: "center",
    fontSize: 60,
    fontWeight: "bold",
    color: Colors.darkBlue,
    marginTop: 80
  },
  plane: {
    alignSelf: "center",
    height: 250,
    marginLeft: 20,
    marginRight: 30,
    marginVertical: 50,
    width: 240
  }
});
