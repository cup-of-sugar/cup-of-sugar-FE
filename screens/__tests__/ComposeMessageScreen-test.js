import React from "react";
import ComposeMessageScreen from "../ComposeMessageScreen";
import { MockedProvider } from "@apollo/react-testing";
import ReactTestRenderer from "react-test-renderer";

it("should render without error if on borrowing path", async () => {
  const props = {
    navigation: { navigate: jest.fn() },
    route: { params: { action: "borrow", userId: "4" } },
  };
  let tree = await ReactTestRenderer.create(
    <MockedProvider mocks={[]}>
      <ComposeMessageScreen {...props} />
    </MockedProvider>
  );

  expect(tree).toMatchSnapshot();
});

it("should render without error if on lending path", async () => {
  const props = {
    navigation: { navigate: jest.fn() },
    route: { params: { action: "lend", userId: "4" } },
  };
  let tree = await ReactTestRenderer.create(
    <MockedProvider mocks={[]}>
      <ComposeMessageScreen {...props} />
    </MockedProvider>
  );

  expect(tree).toMatchSnapshot();
});
