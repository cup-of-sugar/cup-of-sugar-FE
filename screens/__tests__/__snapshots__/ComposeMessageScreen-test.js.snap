// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "padding": 16,
      "paddingTop": 30,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#385A94",
        "borderTopLeftRadius": 7,
        "borderTopRightRadius": 7,
        "padding": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 23,
          "fontWeight": "bold",
          "marginLeft": 10,
          "padding": 7,
          "textAlign": "left",
        }
      }
    >
      To: Joe@tigers.com
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#385A94",
        "padding": 5,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      name="subject"
      onChangeText={[Function]}
      placeholder="Subject..."
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "#fff",
          "borderColor": "#CCCCCC",
          "borderWidth": 1,
          "fontSize": 22,
          "height": 50,
          "paddingHorizontal": 15,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    style={
      Object {
        "borderBottomLeftRadius": 7,
        "borderBottomRightRadius": 7,
        "borderColor": "#385A94",
        "borderTopWidth": 0,
        "borderWidth": 5,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      multiline={true}
      name="body"
      numberOfLines={40}
      onChangeText={[Function]}
      placeholder="Message..."
      rejectResponderTermination={true}
      style={
        Object {
          "fontSize": 22,
          "height": 400,
          "justifyContent": "flex-start",
          "padding": 15,
          "paddingTop": 20,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#0C94E1",
        "borderColor": "#fff",
        "borderRadius": 10,
        "borderWidth": 1,
        "marginHorizontal": 40,
        "marginVertical": 15,
        "opacity": 1,
        "paddingBottom": 20,
        "paddingTop": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 25,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Send Message
    </Text>
  </View>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "color": "red",
        "fontSize": 20,
        "fontWeight": "bold",
        "margin": 10,
      }
    }
  >
    
  </Text>
</View>
`;

exports[`should render component 2`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "padding": 16,
      "paddingTop": 30,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#385A94",
        "borderTopLeftRadius": 7,
        "borderTopRightRadius": 7,
        "padding": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 23,
          "fontWeight": "bold",
          "marginLeft": 10,
          "padding": 7,
          "textAlign": "left",
        }
      }
    >
      To: Joe@tigers.com
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#385A94",
        "padding": 5,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      name="subject"
      onChangeText={[Function]}
      placeholder="Subject..."
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "#fff",
          "borderColor": "#CCCCCC",
          "borderWidth": 1,
          "fontSize": 22,
          "height": 50,
          "paddingHorizontal": 15,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    style={
      Object {
        "borderBottomLeftRadius": 7,
        "borderBottomRightRadius": 7,
        "borderColor": "#385A94",
        "borderTopWidth": 0,
        "borderWidth": 5,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      multiline={true}
      name="body"
      numberOfLines={40}
      onChangeText={[Function]}
      placeholder="Message..."
      rejectResponderTermination={true}
      style={
        Object {
          "fontSize": 22,
          "height": 400,
          "justifyContent": "flex-start",
          "padding": 15,
          "paddingTop": 20,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#0C94E1",
        "borderColor": "#fff",
        "borderRadius": 10,
        "borderWidth": 1,
        "marginHorizontal": 40,
        "marginVertical": 15,
        "opacity": 1,
        "paddingBottom": 20,
        "paddingTop": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 25,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Send Message
    </Text>
  </View>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "color": "red",
        "fontSize": 20,
        "fontWeight": "bold",
        "margin": 10,
      }
    }
  >
    
  </Text>
</View>
`;
